sklearn/linear_model/least_angle.py

```diff
@@ -1074,6 +1074,8 @@ def fit(self, X, y):
         """
         self.fit_path = True
         X, y = check_X_y(X, y, y_numeric=True)
+        X = as_float_array(X, copy=self.copy_X)
```

Comment:
`check_X_y` never upcasts integers:

```
>>> check_X_y([[1], [2]], np.array([1, 2], dtype=np.int32), y_numeric=True, dtype=[np.float64, np.float32])[1].dtype
type('int32')
```

`y_numeric=True` just checks for non-object dtypes in `y`. We could benefit from a `y_dtype=[np.int64, np.int32]` option to replace `y_numeric` at some point but this is outside of the scope of this PR.